storage/scylla_api.c

```diff
@@ -0,0 +1,342 @@
+    while (cass_iterator_next(iterator)) {
+      (*transactionNum)++;
+    }
+    *transaction = malloc((*transactionNum) * sizeof(Transaction));
+    iterator = cass_iterator_from_result(result);
+
+    while (cass_iterator_next(iterator)) {
+      const CassRow* row = cass_iterator_get_row(iterator);
+      get_blob(row, &(((*transaction) + idx)->bundle), "bundle");
+      get_blob(row, &(((*transaction) + idx)->address), "address");
+      get_blob(row, &(((*transaction) + idx)->transaction), "transaction");
+      get_blob(row, &(((*transaction) + idx)->timestamp), "timestamp");
+      get_text(row, &(((*transaction) + idx)->message), "message");
+      cass_value_get_int64(cass_row_get_column_by_name(row, "value"), &(((*transaction) + idx)->value));
+      idx++;
+    }
+
+    cass_result_free(result);
+    cass_iterator_free(iterator);
+  }
+
+  cass_future_free(future);
+  cass_statement_free(statement);
+
+  return rc;
+}
+
+int select_transactions_with_bundle(CassSession* session, const Select_Way select_way,
+                                    const Select_Where** select_where, const size_t select_num) {
+  Transaction* output;
+  int outputNum = 0;
+  CassStatement* statement = NULL;
+  const CassPrepared* select_prepared = NULL;
+
+  if (prepare_query(session, select_query[select_way].query, &select_prepared) == CASS_OK) {
+    statement = cass_prepared_bind(select_prepared);
+    for (size_t i = 0; i < select_num; i++) {
+      statement = cass_prepared_bind(select_prepared);
+      if (select_way == WITH_BUNDLE) {
+        cass_statement_bind_string_by_name(statement, "bundle", ((*select_where) + i)->bundle);
+      } else if (select_way == WITH_BUNDLE_AND_ADDRESS) {
+        cass_statement_bind_string_by_name(statement, "bundle", ((*select_where) + i)->bundle);
+        cass_statement_bind_string_by_name(statement, "address", ((*select_where) + i)->address);
+      }
+      select_from_bundleTable(session, statement, &output, &outputNum);
+      /* serialize to output format here*/
+      SCYLLA_LOG_DEBUG("row = %d\n", outputNum);
+      for (int i = 0; i < outputNum; i++) {
+        SCYLLA_LOG_DEBUG("%s %s %s %s %ld %s\n", output[i].bundle, output[i].address, output[i].transaction,
+                         output[i].message, output[i].value, output[i].timestamp);
+      }
+
+      free(output);
+    }
+    cass_prepared_free(select_prepared);
+  } else {
+    SCYLLA_LOG_ERROR("%s\n", "prepare SELECT query fail");
+    return -1;
+  }
+  return 0;
+}
+
+int scylla_test(char* hosts) {
+  CassCluster* cluster = NULL;
+  CassSession* session = cass_session_new();
+  if (init(cluster, session, hosts) != 0) {
+    return -1;
+  }
+
+  Transaction transactions[] = {
+      {(cass_byte_t*)"2", (cass_byte_t*)"addressHash0", (cass_byte_t*)"transactionHash1", "@#$%^&^%$%^&*",
+       (cass_int64_t)0, (cass_byte_t*)"10"},
+      {(cass_byte_t*)"2", (cass_byte_t*)"addressHash1", (cass_byte_t*)"transactionHash1", ".\.\.\.\32k4j32l432",
```

Comment:
I have no idea about `".\.\.\.\32k4j32l432"`. Can you explain?

Reply:
I am not sure about how a string with undefined escaped sequences will be stored.
It seems to ignore the undefined escaped sequences with such result as below.
....k4j32l432   length 14
![image](https://user-images.githubusercontent.com/3009797/62009593-a058e500-b193-11e9-96dc-042763f3ef82.png)

Reply:
I will remove it.